extensions/resource-deployment/src/services/bigDataClusterDeploymentProfile.ts

```diff
@@ -0,0 +1,256 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export const SqlServerMasterResource = 'master';
+export const DataResource = 'data-0';
+export const HdfsResource = 'storage-0';
+export const ComputeResource = 'compute-0';
+export const NameNodeResource = 'nmnode-0';
+export const SparkHeadResource = 'sparkhead';
+export const ZooKeeperResource = 'zookeeper';
+export const SparkResource = 'spark-0';
+export const HadrEnabledSetting = 'hadr.enabled';
+
+export class BigDataClusterDeploymentProfile {
+	constructor(private _profileName: string, private _bdcConfig: any, private _controlConfig: any) {
+		// TODO: add validation logic for these 2 objects
+	}
```

Comment:
Issue created for tracking this?

Reply:
I would highly recommend creating a type/interface for these too. Ideally you would actually validate the schema when parsing the configs (VS Code does this but I don't think that's available to extensions) but even just having a definition of what you do expect the schema to be and then making all the properties optional and checking appropriately makes it **much** easier to work with. 

---
In reply to: [327726149](https://github.com/microsoft/azuredatastudio/pull/7183#discussion_r327726149) [](ancestors = 327726149)

Reply:
https://github.com/microsoft/azuredatastudio/issues/7344
added to the code